src/utils/EVCUtil.sol

```diff
@@ -26,27 +29,23 @@ abstract contract EVCUtil {
         if (msg.sender == address(evc)) {
             _;
         } else {
-            bytes memory result = evc.call(address(this), msg.sender, 0, msg.data);
-
-            assembly {
-                return(add(32, result), mload(result))
-            }
-        }
-    }
-
-    /// @notice Ensures that the msg.sender is the EVC by using the EVC callback functionality if necessary.
-    /// @dev Optional to use for functions requiring account and vault status checks to enforce predictable behavior.
-    /// @dev If this modifier used in conjuction with any other modifier, it must appear as the first (outermost)
-    /// modifier of the function.
-    /// @dev This modifier is used for payable functions because it forwards the value to the EVC.
-    modifier callThroughEVCPayable() virtual {
-        if (msg.sender == address(evc)) {
-            _;
-        } else {
-            bytes memory result = evc.call{value: msg.value}(address(this), msg.sender, msg.value, msg.data);
+            bytes32 _evcCallSelector = evc.call.selector;
+            address _evc = address(evc);
 
             assembly {
-                return(add(32, result), mload(result))
+                mstore(0, _evcCallSelector) // EVC.call selector
```

Comment:
In the vault, we discussed it may actually be safer to hard-code the selector constant, so it doesn't automatically "work" if the actual evc.call signature changed. Should we do that here too?